watch.go

```diff
@@ -112,38 +141,106 @@ func NewLogWatcherCmd() *cli.Command {
 				return err
 			}
 
-			return WatchForChanges(ctx, cfg, reader, &defaultReporter{log: cmd.log})
+			return WatchForChanges(ctx, cfg, reader, reporter)
 		},
 	}
 }
 
+func checkCompatibleFlags(cCtx cliContext) error {
+	if !cCtx.IsSet(flagLatest) {
+		return nil
+	}
+
+	latestExcludes := []string{flagHorizon, flagSince, flagIDSince}
+
+	for _, excluded := range latestExcludes {
+		if cCtx.IsSet(excluded) {
+			return fmt.Errorf("the %s flag is mutualy exclusive with %s", flagLatest, strings.Join(latestExcludes, ", "))
+		}
+	}
+	return nil
+}
+
 type cliContext interface {
+	IsSet(string) bool
+	Bool(string) bool
 	Duration(name string) time.Duration
 	Timestamp(name string) *time.Time
 	String(name string) string
 	Int(name string) int
 }
 
+// parseHorizon parses a duration string from the command line In accordance
+// with the most common reason for parse failure (specifying a large number), On
+// an error that looks like a range to large issue, we coerce to the maximum
+// hours and ignore the error. Errors that don't contain the marker substring
+// are returned as is.
+func parseHorizon(horizon string) (time.Duration, error) {
+
+	if horizon == horizonAliasMax {
+		return maxHorizon, nil
+	}
+
+	d, err := time.ParseDuration(horizon)
+	if err == nil {
+
+		if d > maxHorizon {
+			return maxHorizon, fmt.Errorf("the maximum supported duration is --horizon=%v, which has the alias --horizon=max. also consider using --latest", maxHorizon)
```

Comment:
minor: should return zero value for first element in tuple if the error is non nil

Reply:
ok will do